### PR TITLE
[GrpcNetClient] Clarify impact to SuppressDownstreamInstrumentation

### PR DESCRIPTION
**Pull request by alanwest**
We should probably decide what we're actually going to do with `SuppressDownstreamInstrumentation`, but for the purpose of getting the next release out, I just wanted to clean up our advisory.

### CODE REVIEW

**Comment by reyang** [comment]
> We should probably decide what we're actually going to do with `SuppressDownstreamInstrumentation`, but for the purpose of getting the next release out, I just wanted to clean up our advisory.

Would this PR description help on the direction? https://github.com/open-telemetry/opentelemetry-dotnet/pull/960#issue-670039806

**Comment by alanwest** [comment]
> Would this PR description help on the direction?

Yes, this is an interesting thought to consider. I believe what you're saying is there's no reason instrumentation needs an SDK dependency to check `otel.suppress_instrumentation` using `RuntimeContext` from `OpenTelemetry.Api` directly. We'd add the check back in the HTTP instrumentation as well.

Does anyone have strong feelings that we pursue this fix prior to releasing 1.7.1 today (#5322)? (@utpilla, @CodeBlanch, @vishweshbankwar)

My opinion is, since this breaking change was introduced last November that we merge this PR with a clear advisory, release 1.7.1 today, and then follow up with this idea.

**Comment by reyang** [comment]
> > Would this PR description help on the direction?
> 
> Yes, this is an interesting thought to consider. I believe what you're saying is there's no reason instrumentation needs an SDK dependency to check `otel.suppress_instrumentation` using `RuntimeContext` from `OpenTelemetry.Api` directly. We'd add the check back in the HTTP instrumentation as well.

Correct, it's just a convention/suggestion (instead of a promise) that anyone can choose to follow (e.g. if there is another telemetry SDK, it can choose to respect this flag as well).

File: src/OpenTelemetry.Instrumentation.GrpcNetClient/CHANGELOG.md [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
-* **Breaking Change** :
+* **Breaking Change**:
+  Please be advised that the
   [SuppressDownstreamInstrumentation](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.GrpcNetClient#suppressdownstreaminstrumentation)
-  option will no longer be supported when used with certain versions of
-  `OpenTelemetry.Instrumentation.Http` package. Check out this
+  option no longer works when used in conjunction with the
+  `OpenTelemetry.Instrumentation.Http` package version `1.6.0` or greater.
+  This is not a result of a change in the `OpenTelemetry.Instrumentation.GrpcNetClient`
+  package therefore this affects versions prior to this release. See this
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.GrpcNetClient/CHANGELOG.md:15]:
```suggestion
  package therefore this also affects versions prior to this release. See this
```

**Comment by utpilla** [comment]
> Does anyone have strong feelings that we pursue this fix prior to releasing 1.7.1 today (#5322)?

No. 

> My opinion is, since this breaking change was introduced last November that we merge this PR with a clear advisory, release 1.7.1 today, and then follow up with this idea.

I'm fine with that.

**Review by utpilla — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
> > > Would this PR description help on the direction?
> > 
> > 
> > Yes, this is an interesting thought to consider. I believe what you're saying is there's no reason instrumentation needs an SDK dependency to check `otel.suppress_instrumentation` using `RuntimeContext` from `OpenTelemetry.Api` directly. We'd add the check back in the HTTP instrumentation as well.
> 
> Correct, it's just a convention/suggestion (instead of a promise) that anyone can choose to follow (e.g. if there is another telemetry SDK, it can choose to respect this flag as well).



> > > Would this PR description help on the direction?
> > 
> > 
> > Yes, this is an interesting thought to consider. I believe what you're saying is there's no reason instrumentation needs an SDK dependency to check `otel.suppress_instrumentation` using `RuntimeContext` from `OpenTelemetry.Api` directly. We'd add the check back in the HTTP instrumentation as well.
> 
> Correct, it's just a convention/suggestion (instead of a promise) that anyone can choose to follow (e.g. if there is another telemetry SDK, it can choose to respect this flag as well).

I believe this is a good option to consider for offering `SuppressDownstreamInstrumentation` as an experimental feature. Experimental because we also need to think about the case when instrumentation will be done natively by HttpClient (starting with `.NET 9.0`) and eventually by Grpc.Net.Client as well.

I am good with going ahead with the release for now.

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**